<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="mav_name" default="cygnus" />
  <arg name="namespace" default="$(arg mav_name)"/>
  <group ns="$(arg namespace)">
    <node name="cygnus_position_controller" pkg="cygnus_control" type="control_altitud" respawn="true" clear_params="true" output="screen">
      <remap from="odometry" to="ground_truth/odometry" />
      <rosparam file="$(find cygnus_control)/resources/$(arg mav_name).yaml" />
      <rosparam file="$(find mav_disturbance_observer)/resources/disturbance_observer_$(arg mav_name)_position_pid.yaml"/>
      <param name="verbose" value="true" />
      <param name="reference_frame" value="world" />
    </node> 

    <node name="PID_attitude_controller" pkg="mav_lowlevel_attitude_controller" type="mav_pid_attitude_controller_node" respawn="true" clear_params="true" output="screen">
      <remap from="odometry" to="ground_truth/odometry" />
      <rosparam file="$(find cygnus_control)/resources/PID_attitude_$(arg mav_name).yaml" />
    </node>
  </group>
</launch>
